personcreator.rb

```diff
@@ -0,0 +1,39 @@
+class PersonCreator
+  def initialize(library)
+    @library = library
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    choice = gets.chomp
+    if choice == '1'
+      create_student_from_input
+    elsif choice == '2'
+      create_teacher_from_input
+    else
+      puts 'Invalid choice'
+    end
+  end
+
+  def create_student_from_input
+    print 'Age: '
```

Comment:
- [ ] I can see a lot of input requests in your methods that create, I will like to remind you of YAGNI and KISS which you learned [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid), it would be better to have all requests for input and input validation placed in the main.rb file so that it would satisfy the principles earlier pointed out.